### DISCUSSION TITLE
How to enable OntapS3 and OntapS3SVM Harvest templates

### DISCUSSION THREAD

**Discussion by Hardikl** [NetApp/harvest#3137] [Show and tell]
#### This changes would be needed for enabling OntapS3 and OntapS3SVM counters.

<details>
<summary>Native</summary>

> :warning: **Note:**  
Below steps are applicable for native install. 

<details>
<summary>OntapS3 Rest Collector</summary>

## Step 1: Create `custom.yaml`

Let's assume Harvest is installed in `/opt/harvest` directory. Adjust paths as needed.

Navigate to the `rest` directory:

```bash
cd /opt/harvest/conf/rest
```

Create custom.yaml:

```bash
vi custom.yaml
```

Update the file with the following content:

```yaml
objects:
  OntapS3:  ontap_s3.yaml
```

## Step 2: Restart Harvest pollers

```
bin/harvest restart
```
</details>

<details>
<summary>OntapS3SVM RestPerf Collector</summary>
       
## Step 1: Create `custom.yaml`

Let's assume Harvest is installed in `/opt/harvest` directory. Adjust paths as needed.

Navigate to the `rest` directory:

```bash
cd /opt/harvest/conf/restperf
```

Create custom.yaml:

```bash
vi custom.yaml
```

Update the file with the following content:

```yaml
objects:
  OntapS3SVM:  ontap_s3_svm.yaml
```

## Step 2: Restart Harvest pollers

```
bin/harvest restart
```
</details>

<details>
<summary>OntapS3SVM ZapiPerf Collector</summary>
       
## Step 1: Create `custom.yaml`

Let's assume Harvest is installed in `/opt/harvest` directory. Adjust paths as needed.

Navigate to the `rest` directory:

```bash
cd /opt/harvest/conf/zapiperf
```

Create custom.yaml:

```bash
vi custom.yaml
```

Update the file with the following content:

```yaml
objects:
  OntapS3SVM:  ontap_s3_svm.yaml
```

## Step 2: Restart Harvest pollers

```
bin/harvest restart
```
</details>

</details>

<details>
<summary>NABox4</summary>

> :warning: **Note:**  
**Prerequisite:** ssh to NABox via [admin](https://nabox.org/documentation/configuration/) 

<details>
<summary>OntapS3 Rest Collector</summary>

## Step 1: Create `custom.yaml`

Navigate to the `rest` directory:

```bash
cd /etc/nabox/harvest/user/rest
```

Create custom.yaml:

```bash
vi custom.yaml
```

Update the file with the following content:

```yaml
objects:
  OntapS3: ontap_s3.yaml
```

## Step 2: Restart NABox containers

```
dc restart
```
</details>

<details>
<summary>OntapS3SVM RestPerf Collector</summary>
      
## Step 1: Create `custom.yaml`

Navigate to the `rest` directory:

```bash
cd /etc/nabox/harvest/user/restperf
```

Create custom.yaml:

```bash
vi custom.yaml
```

Update the file with the following content:

```yaml
objects:
  OntapS3SVM: ontap_s3_svm.yaml
```

## Step 2: Restart NABox containers

```
dc restart
```
</details>

<details>
<summary>OntapS3SVM ZapiPerf Collector</summary>
      
## Step 1: Create `custom.yaml`

Navigate to the `rest` directory:

```bash
cd /etc/nabox/harvest/user/zapiperf
```

Create custom.yaml:

```bash
vi custom.yaml
```

Update the file with the following content:

```yaml
objects:
  OntapS3SVM: ontap_s3_svm.yaml
```

## Step 2: Restart NABox containers

```
dc restart
```
</details>

</details>

**Comment by mgollo** [comment]
On NABox 4.0.9 with Harvest 25.02.0, the correct content of /etc/nabox/harvest/user/restperf/custom.yaml in order to enable the OntapS#SVM HArvest template is as follows:
```
objects:
  OntapS3SVM: "ontap_s3_svm.yaml"
```

When defining OntapS3SVM as an array as in your example, harvest will throw an error like this:
```
ERROR source=havrest.go:49 msg="unable to merge config" error="yaml: unmarshal errors:\n  line 3: cannot unmarshal !!seq into string"
```

**Comment by Hardikl** [comment]
Thanks @mgollo ,

We have checked and updated the docs as per `key: value` for custom.yaml.
